### ISSUE TITLE
Class-attribute checking only recognises `.. attribute :: name`, which is invalid reStructuredText

### ISSUE THREAD

**Issue by shadowk29** [jsh9/pydoclint#304]
`pydoclint` 0.9.1 · `docstring_parser_fork` 0.0.16 · Python 3.12 · `style = "sphinx"`

## Summary

In Sphinx/reST style, `--check-class-attributes` recognises class-attribute
documentation **only** when the directive is written `.. attribute :: name`, with a
space before the `::`. That form is not a valid reStructuredText directive — docutils
requires `.. name:: arguments`, and with the extra space it parses the line as a
**comment**, so Sphinx renders nothing for it.

The correctly-written directive `.. attribute:: name` is not recognised at all, and
neither is any canonical reST field form (`:ivar:`, `:cvar:`, `:var:`). The net effect
is inverted from what you'd want: pydoclint accepts exactly the form Sphinx ignores, and
rejects the form Sphinx renders.

This also means `checking_class_attributes.html` is currently telling users to write
invalid reST — the Sphinx example there is `.. attribute :: name`.

## Reproduction

```python
# repro.py
class Pet:
    """
    A pet.

    .. attribute:: name
        :type: str

        Name of my pet

    :param owner: Owner.
    :type owner: str
    """

    name: str = "dog"

    def __init__(self, owner: str) -> None:
        self.owner = owner
```

```console
$ pydoclint --style=sphinx repro.py
repro.py
    1: DOC601: Class `Pet`: Class docstring contains fewer class attributes than actual class attributes.
    1: DOC603: Class `Pet`: Class docstring attributes are different from actual class attributes.
       Attributes in the class definition but not in the docstring: [name: str].
```

Adding a single space (`.. attribute :: name`) makes it pass. Nothing else changes.

**Expected:** the valid directive `.. attribute:: name` is recognised; ideally the
spaced form keeps working too, for backwards compatibility with docstrings written
against the current documentation.

## Why this is easy to miss

`--skip-checking-short-docstrings` defaults to `True`, and a class docstring whose only
"section" is the unrecognised directive parses as a short docstring, so the check is
skipped entirely. Take the same class with the `:param owner:` block and `__init__`
removed, leaving only the description and the directive:

```console
$ pydoclint --style=sphinx repro_short.py
# clean

$ pydoclint --style=sphinx --skip-checking-short-docstrings=False repro_short.py
    1: DOC601: Class `Pet`: Class docstring contains fewer class attributes than actual class attributes.
    1: DOC603: Class `Pet`: Class docstring attributes are different from actual class attributes.
```

So the false positive only surfaces once the class docstring also documents `__init__`
parameters — which is precisely the case the documentation page tells users to write.

## Root cause

`docstring_parser_fork/rest_attr_parser.py` hardcodes the spaced literal in two places:

```python
# parse_attributes()
if stripped_line.startswith(".. attribute ::"):

# parse_attribute_block()
if stripped_line.startswith(".. attribute ::"):
    name = stripped_line[len(".. attribute ::"):].strip()
```

Nothing else populates `Docstring.attrs` for reST style, and pydoclint's `Doc.attrList`
reads `self.parsed.attrs` — so with a correctly-written directive the documented-attribute
list is always empty.

Confirmed directly against the parser:

```python
from docstring_parser import parse, DocstringStyle

def attrs(doc):
    return [(a.arg_name, a.type_name) for a in parse(doc, DocstringStyle.REST).attrs]

attrs(".. attribute :: name\n    :type: str\n\n    Name.\n")  # [('name', 'str')]  <- invalid reST
attrs(".. attribute:: name\n    :type: str\n\n    Name.\n")   # []                 <- valid reST
attrs(":ivar name: Name.\n:vartype name: str\n")              # []
attrs(":cvar name: Name.\n:vartype name: str\n")              # []

`:ivar:`/`:cvar:`/`:var:` are parsed — they appear in `Docstring.meta` with the right
keys — they're just never promoted to `DocstringAttr`, so `.attrs` stays empty.

## Suggested fix

Match the directive with the `::` optionally preceded by whitespace, and derive the name
from the match rather than by slicing a fixed-length literal, e.g.

```python
ATTRIBUTE_DIRECTIVE = re.compile(r"^\.\.\s+attribute\s*::\s*(?P<name>.+)$")
```

That accepts both spellings, so no existing docstring breaks. Recognising
`:ivar:`/`:cvar:`/`:var:` as attributes would be a natural follow-up, since those are the
forms most reST codebases already use, but it isn't needed to fix this.

The documentation example on `checking_class_attributes.html` should also change to
`.. attribute:: name` once the parser accepts it.

## Which repo?

Filing here on `pydoclint` because the documentation page that prescribes the invalid
syntax is pydoclint's, and the user-visible symptom is DOC601/DOC603. The code change
itself belongs in `jsh9/docstring_parser_fork` (`rest_attr_parser.py`) — happy to move
the issue there if you'd prefer.

## Workaround

For anyone hitting this before it's fixed: `check-class-attributes = false`. Under sphinx
style the check can currently only fire against docstrings that are wrong, so nothing of
value is lost until the parser is fixed.

Every claim in it is verified against the installed packages rather than inferred — the reproduction, the parser table, the skip-checking-short-docstrings masking, and the suggested regex (which I tested against both spellings plus .. attributes:: and .. note:: to confirm it doesn't over-match).
